src/ucs/sys/sock.c

```diff
@@ -124,6 +124,15 @@ static const char *ucs_socket_getname_str(int fd, char *str, size_t max_size)
                             str, max_size);
 }
 
+static ucs_status_t ucs_socket_check_errno(int io_errno)
+{
+    if ((io_errno == EINTR) || (io_errno == EAGAIN) || (io_errno == EWOULDBLOCK)) {
```

Comment:
minor:
IMO `(io_errno == EAGAIN) || (io_errno == EWOULDBLOCK)` better to check first since it's "likely" path for empty socket, `EINTR` is heavier path and can be checked last